docs(testing): re-organize testing up to allow adding more frameworks (#1344)
ionic-team/stencil-site: docs/testing/01-overview.md

together. For example, when `my-component` has the `X` attribute, the child component then renders the text `Y`, and
expects to receive the event `Z`.

## Library Support

Stencil currently supports the following tools for testing components:

> - [__Stencil Test Runner__](./stencil-testrunner/01-overview.md): a built-in test runner based on Jest for unit and end-to-end testing with Puppeteer to run within an actual browser in order to provide more realistic results.

Reviewer: This link doesn't look obviously like a link on the site:

<img width="579" alt="Screenshot 2024-02-07 at 1 27 45 PM" src="https://github.com/ionic-team/stencil-site/assets/6207644/8422b186-edf6-4d98-b19f-1b75291ec79b">

without the bold its "linky-ness" is visually obvious:

<img width="447" alt="Screenshot 2024-02-07 at 1 28 22 PM" src="https://github.com/ionic-team/stencil-site/assets/6207644/9f9fdf1e-a92a-4ccb-8557-fa506623b2f3">

Might be a styling issue, but I think unless we're empowered to change the theme we might want to do something like this:

```suggestion
- [Stencil Test Runner](./stencil-testrunner/01-overview.md): a built-in test runner based on Jest for unit and end-to-end testing with Puppeteer to run within an actual browser in order to provide more realistic results.
```

Author: Yeah, this looks way better. Thanks!

:::info

> We are actively working supporting a wider range of testing tools, including Playwright and WebdriverIO. Stay tuned for updates!

Reviewer 2: Minor grammatical fix

```suggestion
We are actively working to support a wider range of testing tools, including Playwright and WebdriverIO. Stay tuned for updates!
```

> We are actively working supporting a more wider range of testing tools, including Playwright and WebdriverIO. Stay tuned for updates!

Reviewer 3: Non-blocking grammar nit
```suggestion
We are actively working supporting a wider range of testing tools, including Playwright and WebdriverIO. Stay tuned for updates!
```
